plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -141,7 +250,62 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 				);
 			}
 		}
+	}
 
+	/**
+	 * Adds a preload link for a <picture> element.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_Tag_Visitor_Context                                                                                                                    $context Tag visitor context.
+	 * @param string                                                                                                                                    $xpath   XPath of the element.
+	 * @param array{srcset?: string|true|null,sizes?: string|true|null,type?: string|true|null,media?: string|true|null,crossorigin?: string|true|null} $source  Collected sources from the <source> elements.
+	 */
+	private function add_preload_link_for_picture( OD_Tag_Visitor_Context $context, string $xpath, array $source ): void {
```

Comment:
I think this logic could be put inside `process_picture()`, unless there is an opportunity to refactor it so that it can be reused both for `PICTURE` and `IMG`.